genie-web/src/main/java/com/netflix/genie/web/agent/services/impl/AgentJobServiceImpl.java

```diff
@@ -133,8 +138,21 @@ public void handshake(
     public Map<String, String> getAgentProperties(
         @Valid final AgentClientMetadata agentClientMetadata
     ) {
-        // TODO: currently NOOP. Implement.
-        return Maps.newHashMap();
+        final HashSet<Tag> tags = Sets.newHashSet(
+            Tag.of(AGENT_VERSION_METRIC_TAG_NAME, agentClientMetadata.getVersion().orElse("null")),
+            Tag.of(AGENT_HOST_METRIC_TAG_NAME, agentClientMetadata.getHostname().orElse("null"))
```

Comment:
I think this is a bad idea for cardinality. It makes sense in theory but in reality we're going to get a lot of UUID's for host names on containers

Reply:
Indeed, i added this for consistency with the rest of this class, but i meant to then add a commit that removes all hostname tags).
I'll add it in this same PR.